### ISSUE TITLE
Topic operator should handle change of underlying k8s topic

### ISSUE THREAD

**Issue by sknot-rh** [strimzi/strimzi-kafka-operator#5674]
From slack discussion:

> After a migration we have two KafkaTopic CRs related to the same Kafka topic. The new one is not picked up by the Topic Operator, obviously, with "Topic 'a.b.c' is already managed via KafkaTopic 'a-b-c' it cannot be managed via the KafkaTopic 'd-e-f'" message. Now, we would like to have only one KafkaTopic CR related to a given topic, and we want to remove the older KafkaTopic CR. However, after removing the older CR, although the error message disappears from the newer CR, it doesn't seem to be fully picked up by the Topic Operator - for example, if we change the configuration of the topic in the newer CR, the Kafka topic does not get updated.

**Comment by rlyzwa** [comment]
Hello, I'm the one who asked this question on the Slack.

The following solution works for us on Strimzi **0.18**. 

Let's say KafkaTopic CR 'my-test-topic-1' was created on K8s and created the topic 'my.test.topic'. Then, 'my-test-topic-2', referencing also 'my.test.topic' was created. While 'my-test-topic-1' has the correct status, 'my-test-topic-2' contains the error message coming from the Topic Operator: "Topic 'my.test.topic' is already managed via KafkaTopic 'my-test-topic-1' it cannot be managed via the KafkaTopic 'my-test-topic-2'".

What we want to achieve is to delete the 'my-test-topic-1' and make sure that Strimzi uses 'my-test-topic-2' to manage 'my.test.topic'. Since we don't want to lose the topic data, we need to disable first the automatic topic deletion in Strimzi's Kafka CR. Then, when brokers are updated, we can start our procedure.

The Topic Operator keeps information about topics in memory, but it also stores it in the topic store. In recent versions of Strimzi topic store is persisted in the internal Kafka topic. However, in our version (0.18), ZooKeeper is used for that purpose. So the idea of the procedure is to make sure that both KafkaTopic CRs and the topic store metadata in the ZK are aligned and then delete the older CR.

The steps are following:

1. Make sure that 'my-test-topic-2' contains exactly the same 'spec.config' section as 'my-test-topic-1'. This configuration is also normally reflected in the ZK, which can be checked with:

```
/bin/zookeeper-shell.sh localhost:2181 get /strimzi/topics/my.test.topic

Connecting to localhost:2181

WATCHER:

WatchedEvent: state:SyncConnected type:None path:null
{"map-name":"my-test-topic-1","topic-name":"my.test.topic","partitions":1,"replicas":1,"config":{"retention.ms":"4783743874"}}
```

2. Update the ZK entry, so it contains 'my-test-topic-2' in map-name:

```
/bin/zookeeper-shell.sh localhost:2181 set /strimzi/topics/my.test.topic '{"map-name":"my-test-topic-2","topic-name":"my.test.topic","partitions":1,"replicas":1,"config":{"retention.ms":"4783743874"}}'
```

3. Execute the following commands one after another:

```
oc scale --replicas=0 deployment/main-entity-operator
oc delete kafkatopic "my-test-topic-1"
```

4. Once the Entity Operator (and the Topic Operator) are up, 'my.test.topic' should be managed by 'my-test-topic-2'. You can test by modifying something in 'spec.config' section of the KafkaTopic CR.

(Deleting the old KafkaTopic CR when the Topic Operator was not running seemed unavoidable for us. Otherwise, probably because of in-memory state kept by the Topic Operator, an entry was re-created in ZK, with "map-name" equal to the name of the topic: "my.test.topic")

**Comment by sknot-rh** [comment]
I just tried to reproduce this with the latest Strimzi version. Your bug seems to be fixed in 0.22 by changing storing topic mapping from ZK to Kafka topic (`strimzi-store-topic) https://github.com/strimzi/strimzi-kafka-operator/pull/3405. 
Could you please confirm?
